Add readme

fbouliane opened the pull request.

fbouliane committed:
Add readme
Makes it easier to understand context

godp1301 commented on README.md:
```diff
@@ -0,0 +1,73 @@
+The normal behavior of redlock-py is to `return false` on a existing lock. The mainstream approach is to 
+poll the lock until acquisition. This process has two downsides :
+
+1. It doesn't warranty that the first requester will be the first to get the lock.
```
IMO warranty is not a verb so it can't be used likewise. Perhaps guarantee would be better...

fbouliane committed:
Fix typos in README.md

Change guarantee wording.
Remove duplicate sentence that added nothing.

fbouliane committed:
Make sure the test simulates real-world.

Make the connector is global for every lock request.
Before, the connector was local to a thread, with independant redis mock.
Therefore, no concurrent testing was done.

fbouliane committed:
Introducing entropy in get_lock_and_register method.

Ensure that the order of locking request has a random component.

fbouliane committed:
Make sure the test can reproduce a failure.

Adding a loop testing over 100 iterations habitually reproduce the bug
This test is too long to run every time but can be helpful.

JoProvost commented:
Looks good to me :+1:

idjaw commented on README.md:
```diff
@@ -0,0 +1,71 @@
+Redlock-Fifo makes [Redlock-py](https://github.com/SPSCommerce/redlock-py) usable in a multi-threaded environment.
+
+This project makes sure that the requester will acquire a lock according to the order it is asked for. It also 
+adds the ability to extends a lock without having to release the lock.
```
s/extends/extend/

idjaw commented on README.md:
```diff
@@ -0,0 +1,71 @@
+
+To prevent [starvation](https://en.wikipedia.org/wiki/Starvation_%28computer_science%29), we add a simple FIFO queue 
+implemented in Redis, managing access to the resource. This queue allows us to guarantee access to a resource and guarantee
+ that the lock will be executed in order of lock request.
```
remove the second guarantee.

fbouliane committed:
Fix typos.

idjaw commented:
:+1:
